app/queries/admin/users_query.rb

```diff
@@ -46,29 +48,40 @@ def self.filter_organization_memberships(relation:, organizations:)
     #       query per role is inadequate.
     #
     # @see https://github.com/RolifyCommunity/rolify/blob/0c883f4173f409766338b9c6dfc64b0fc8ec8a52/lib/rolify/finders.rb#L26-L32
-    def self.filter_roles(relation:, roles:, role_map: Constants::Role::SPECIAL_ROLES_LABELS_TO_WHERE_CLAUSE)
+    def self.filter_roles(relation:, roles:, role_map: Constants::Role::ALL_ROLES_LABELS_TO_WHERE_CLAUSE)
       conditions = []
       values = []
 
       # Assemble the conditions and positional parameter values
       roles.each do |role|
         where_clause = role_map.fetch(role)
-        resource_type = where_clause.fetch(:resource_type, nil)
-        condition = "(#{Role.table_name}.name = ? AND #{Role.table_name}.resource_id IS NULL"
-        values << where_clause.fetch(:name)
 
-        # We need to use `IS NULL` instead of `= NULL` as those are different meanings.
-        if resource_type.nil?
-          condition += " AND #{Role.table_name}.resource_type IS NULL"
+        # TODO: When we filter by "Good standing" status, we want to filter by
+        # NOT any of [trusted, warned, suspended, comment_suspended]
+        if where_clause == Constants::Role::GOOD_STANDING_STATUS
+          conditions << "(#{Role.table_name}.name NOT IN (?))"
+          values << Constants::Role::FOLKS_WITH_THESE_ROLES_ARE_NOT_COUNTED_AS_IN_GOOD_STANDING
         else
```

Comment:
@jeremyf  Now that things are "unpausing" I caught up on these changes for filtering by "Good standing" - unfortunately, it doesn't _quite_ work as-is, because users with "special roles" like admin, tag moderator, etc cause a user to slip through this condition.

Nothing is simple! Anyway, thank you for looking at this - it's been decided that we'll descope filter by "good standing" due to the complexity, so I'm going to go ahead and roll back some of the changes we've both made and get this up for review without 'good standing'